### PR TITLE
Update hero ability and item building for 7.32e

### PR DESCRIPTION
**Pull request by ArcherWayne**
The patch for 7.32e is ready for reviewing. Please review this update.

### CODE REVIEW

**Comment by adamqqqplay** [comment]
It would be better to use the imperative sentence for the commit summary.

File: util/BotNameUtility.lua [hunk 1]
```diff
@@ -218,6 +218,56 @@ local dota2team = {
 		};
 		['sponsorship'] = '';
 	},
+	-- Ti 10
```
Comment (adamqqqplay) [inline util/BotNameUtility.lua:221]: Changes in this file could be split into another commit.

File: item_purchase_nevermore.lua [hunk 2]
```diff
@@ -6,6 +6,7 @@ local ItemPurchaseSystem = dofile(GetScriptDirectory() .. "/util/ItemPurchaseSys
 
 local ItemsToBuy =
 {
+	"item_tango",
```
Comment (adamqqqplay) [inline item_purchase_nevermore.lua:9]: Consumables don't seem to need to be added to each hero's buy list, it should be automatically bought in the public item purchasing module.

File: item_purchase_abaddon.lua [hunk 3]
```diff
@@ -6,21 +6,24 @@ local ItemPurchaseSystem = dofile(GetScriptDirectory() .. "/util/ItemPurchaseSys
...
-	"item_vladmir", --祭品7.21
-	"item_mekansm", --梅肯
-	"item_guardian_greaves", --卫士胫甲
-	"item_pipe", --笛子
-	"item_helm_of_the_overlord", --level 2 helm dominator 7.30
-	"item_ultimate_scepter", --蓝杖
-	"item_ultimate_scepter_2",
-	"item_lotus_orb" --清莲宝珠
+	"item_phase_boots",
+    "item_radiance",
```
Comment (adamqqqplay) [inline item_purchase_abaddon.lua:15]: There seems to be a problem with the indentation of the code, it would be best to use a formatter to reformat all modified files.

File: ability_item_usage_furion.lua [hunk 4]
```diff
@@ -49,12 +49,12 @@ local AbilityToLevelUp = {
 
 local TalentTree = {
```
Comment (adamqqqplay) [inline ability_item_usage_furion.lua:50]: Why are there 8 talent options here? 
I see, it should be the selection order after level 25. I don't know if other heroes will choose talents after level 25.

File: hero_selection.lua [hunk 5]
```diff
@@ -766,7 +766,7 @@ function AllPickLogic()
 					(GetSelectedHeroName(id) == "" or GetSelectedHeroName(id) == nil))
 			then
 				if (randomTime == 0) then
-					randomTime = RandomInt(10, 12);
+					randomTime = RandomInt(1, 2);
```
Comment (adamqqqplay) [inline hero_selection.lua:769]: Maybe this is too fast?

File: item_purchase_nevermore.lua (hunk 2 above)
Comment (ArcherWayne) [reply item_purchase_nevermore.lua:9]: It was discovered that if the consumables were not added to the hero's buy list the hero will not buy the said consumables. Perhaps the public item purchasing module did not work properly?

File: hero_selection.lua (hunk 5 above)
Comment (ArcherWayne) [reply hero_selection.lua:769]: 10 to 12 seconds would make players wait for around one minute. It could be better if bots can choose faster.

File: ability_item_usage_furion.lua (hunk 4 above)
Comment (ArcherWayne) [reply ability_item_usage_furion.lua:50]: The furion code is written in this way. I really don't know why. 

It was playtested that furion and other heroes will choose talents after level 25 properly.

File: item_purchase_abaddon.lua (hunk 3 above)
Comment (ArcherWayne) [reply item_purchase_abaddon.lua]: The indentation problems in this commit had been solved.

File: item_purchase_nevermore.lua (hunk 2 above)
Comment (adamqqqplay) [reply item_purchase_nevermore.lua:9]: Maybe, we could fix it later.

File: ability_item_usage_furion.lua (hunk 4 above)
Comment (adamqqqplay) [reply ability_item_usage_furion.lua:50]: That's interesting.

**Review by adamqqqplay — APPROVED** [review]
